pipeline.yaml

```diff
@@ -0,0 +1,94 @@
+resources:
+  - name: source
+    type: git
+    source:
+      uri: ((git_source))
+      private_key: ((privatekey))
+      branch: ((git_branch))
+      skip_ssl_verification: ((git_skip_ssl_verification))
+      # This enables triggering on tag changes that match the glob pattern
+      # tag_filter: "*.*.*"
+
+  - name: image-bb-gitversion-tool
+    type: docker-image
+    source:
+      repository: elbb/bb-gitversion
+      tag: ((bb_gitversion_version))
+
+  - name: s3-gitversion
+    type: s3
+    source:
+      bucket: ((bb_name))
+      endpoint: ((s3_endpoint))
+      # at least one regex capture group is needed here
+      regexp: version/gitversion-(.*).tar.gz
+      access_key_id: ((s3_access_key))
+      secret_access_key: ((s3_secret_key))
+      disable_ssl: ((s3_skip_ssl_verification))
+
+  - name: image-aws-cli
+    type: docker-image
+    source:
+      repository: amazon/aws-cli
+      tag: latest
+
+jobs:
+  - name: Generate gitversion and put it on S3
+    public: true
+    plan:
+      - in_parallel:
+          - get: image-bb-gitversion-tool
+          - get: source
+            trigger: true
+          - get: image-aws-cli
+      - task: Generate version
+        image: image-bb-gitversion-tool
+        config:
+          platform: linux
+          inputs:
+            - name: source
+          outputs:
+            - name: gen
+          run:
+            path: /bin/bash
+            args:
+              - -exc
+              - |
+                entrypoint.sh && 
+                jq empty gen/json/gitversion.json &&
+                cat gen/json/gitversion.json && 
+                if [ "((git_branch))" == "master" ]; then 
+                  cat gen/plain/SemVer > gen/version
+                else 
+                  cat gen/plain/InformationalVersion > gen/version
+                fi &&
+                sed 's/+/-/g' gen/version > gen/version &&
```

Comment:
doesnt work in my concourse env. the file gen/version is empty afterwards. I had to change the line to 
```
sed -i 's/+/-/g' gen/version &&
```

Reply:
modified